packages/devtools_extensions/lib/src/template/_simulated_devtools_environment/connection_ui/_dtd_connect.dart

```diff
@@ -0,0 +1,136 @@
+      disconnectedLabel: 'Dart Tooling Daemon Connection',
+      disconnectedHint: '(e.g., ws://127.0.0.1:57624)',
+      // TODO(kenz): this needs handling for bad input.
+      onConnect: (value) async {
+        value = value.trim();
+        if (value.startsWith('127.0.0.1')) {
+          value = 'ws://$value';
+        }
+        await onConnectionChange(value);
+        if (dtdManager.hasConnection) {
+          simController.logInfoEvent(
+            'Connected DTD to ${dtdManager.uri?.toString()}',
+          );
+        } else {
+          simController.logInfoEvent('Failed to connect DTD to $value');
+        }
+      },
+      onDisconnect: () async {
+        await onConnectionChange(null);
+        simController.logInfoEvent('Disconnected from DTD');
+      },
+      currentConnection: () => dtdManager.uri?.toString() ?? '--',
+      help: const DtdHelp(),
+    );
+  }
+}
+
+class DtdHelp extends StatelessWidget {
+  const DtdHelp({super.key});
+
+  static const dtdStartCommand = 'dart tooling-daemon --disable-secrets';
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return DefaultTextStyle(
+      style: theme.regularTextStyle,
+      child: Column(
+        children: [
+          RichText(
+            text: TextSpan(
+              style: theme.regularTextStyle,
+              text:
+                  'If your DevTools extension interacts with the Dart Tooling '
+                  'Daemon (DTD) through',
+              children: [
+                TextSpan(
+                  text: ' dtdManager ',
+                  style: theme.boldTextStyle,
+                ),
+                const TextSpan(
+                  text: 'then you will need to start a local instance of DTD '
+                      'to debug these features in the simulated environment. '
+                      'To start DTD locally:\n\n1. Run the following command '
+                      'from your terminal:',
+                ),
+              ],
+            ),
+          ),
+          Padding(
+            padding: const EdgeInsets.symmetric(vertical: defaultSpacing),
+            child: Row(
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                const SelectableText(dtdStartCommand),
+                const SizedBox(width: defaultSpacing),
+                DevToolsButton.iconOnly(
+                  icon: Icons.content_copy,
+                  onPressed: () async {
+                    await Clipboard.setData(
+                      const ClipboardData(text: dtdStartCommand),
+                    );
+                  },
+                ),
+              ],
+            ),
+          ),
+          RichText(
+            text: TextSpan(
+              style: theme.regularTextStyle,
+              text: '2. This will output text to the command line:',
+              children: [
+                TextSpan(
+                  text: ' "The Dart Tooling Daemon is listening on '
```

Comment:
Putting behind `--machine` sounds good. Although, is it a problem to ask users to copy from the printed JSON instead of from a more friendly string? They just need access to the URI so they can copy it, and since the URI is in the printed output (which it looks like it will be), it may be overkill to add another flag just for this niche case. WDYT?